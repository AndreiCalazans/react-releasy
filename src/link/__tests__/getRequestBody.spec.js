import createRequest from '../../../test/fixtures/relay/createRequest';
import createUploadables from '../../../test/fixtures/relay/createUploadables';
import createVariables from '../../../test/fixtures/relay/createVariables';

import getRequestBody from '../getRequestBody';

it('should create default request body', () => {
  const request = createRequest();
  const variables = createVariables();

  const body = getRequestBody(request, variables);

  expect(body).toMatchSnapshot();
});

it('should create request body for uploadables', () => {
  const request = createRequest();
  const variables = createVariables();
  const uploadables = createUploadables();

  const body = getRequestBody(request, variables, uploadables);

  expect(body).toMatchSnapshot();
});
