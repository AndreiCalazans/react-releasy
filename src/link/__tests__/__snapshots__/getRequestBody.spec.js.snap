// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create default request body 1`] = `"{\\"query\\":\\"query meQuery {\\\\n  me {\\\\n    id\\\\n  }\\\\n}\\",\\"variables\\":{}}"`;

exports[`should create request body for uploadables 1`] = `
FormData {
  Symbol(impl): FormDataImpl {
    "_entries": Array [
      Object {
        "name": "query",
        "value": "query meQuery {
  me {
    id
  }
}",
      },
      Object {
        "name": "variables",
        "value": "{}",
      },
      Object {
        "name": "image0",
        "value": "[object Object]",
      },
    ],
    Symbol(wrapper): [Circular],
  },
}
`;
